Realm/RLMObjectBase.mm

```diff
@@ -133,24 +106,48 @@ + (RLMObjectSchema *)sharedSchema {
     return RLMSchema.sharedSchema[self.className];
 }
 
-- (NSArray *)linkingObjectsOfClass:(NSString *)className forProperty:(NSString *)property {
-    if (!_realm) {
+void RLMObjectBaseSetRealm(__unsafe_unretained RLMObjectBase *object, __unsafe_unretained RLMRealm *realm) {
```

Comment:
Annotating every pointer with `REALM_NULLABLE`/`REALM_NONNULL` and building with a 6.3 beta would be a good idea that'll probably catch at least one mismatch between our expectations and implementation.